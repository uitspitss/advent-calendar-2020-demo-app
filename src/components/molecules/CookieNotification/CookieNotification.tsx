import React, { FC } from 'react';
import tw, { styled } from 'twin.macro';
import { keyframes } from '@emotion/react';
import CookieConsent from 'react-cookie-consent';

export type CookieNotificationProps = {
  /**
   * class for twin.macro
   */
  className?: string;
  /**
   * no children
   */
  children?: never;
  /**
   * whether debug
   */
  debug?: boolean;
};

type Props = {
  handleDecline?: () => void;
} & CookieNotificationProps;

const Component: FC<Props> = (props) => {
  const { className, debug = false } = props;

  return (
    <CookieConsent
      disableStyles
      flipButtons
      location="top"
      containerClasses={className}
      buttonClasses="acceptButton"
      // enableDeclineButton
      // declineButtonClasses="declineButton"
      // onDecline={handleDecline}
      debug={debug}
    >
      This website uses cookies to enhance the user experience.
    </CookieConsent>
  );
};

const slideIn = keyframes`
  from {
    transform: translateY(-100%);
  }
  to {
    transform: translateY(0);
  }
`;

const StyledComponent = styled(Component)`
  animation: ${slideIn} 0.5s ease;

  ${tw`flex flex-wrap w-full justify-between items-center bg-info-500 text-white text-sm font-bold px-4 py-3 transition duration-500 ease-in-out`}

  & > div {
    & > .acceptButton {
      ${tw`bg-secondary-500 hover:bg-secondary-700 text-white font-bold py-2 px-4 mr-2 border-none cursor-pointer`}
    }
    & > .declineButton {
      ${tw`bg-transparent no-underline hover:underline text-white font-bold py-2 px-4 border-none cursor-pointer`}
    }
  }
`;

/**
 * cookie の利用確認用 notification
 */
export const CookieNotification: FC<CookieNotificationProps> = (props) => {
  const handleDecline = () => {
    // Cookies.remove('_ga');
    // Cookies.remove(`_ga_${GA_TRACKING_ID}`);
  };

  return <StyledComponent {...props} {...{ handleDecline }} />;
};
