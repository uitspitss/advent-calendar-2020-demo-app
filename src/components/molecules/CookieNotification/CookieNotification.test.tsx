import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import { CookieNotification } from './CookieNotification';

describe('molecules/CookieNotification', () => {
  it('Snap Shot', () => {
    const tree = renderer.create(<CookieNotification />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Behavior', () => {
    beforeEach(() => {
      render(<CookieNotification />);
    });

    test.todo('something of test');
  });
});
