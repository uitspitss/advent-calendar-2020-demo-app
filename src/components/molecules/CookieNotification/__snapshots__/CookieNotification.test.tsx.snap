// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/CookieNotification Snap Shot 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: translateY(-100%);
    -ms-transform: translateY(-100%);
    transform: translateY(-100%);
  }

  to {
    -webkit-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }
}

.emotion-0 {
  -webkit-animation: animation-0 0.5s ease;
  animation: animation-0 0.5s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --bg-opacity: 1;
  background-color: rgba(66,153,225,var(--bg-opacity));
  --text-opacity: 1;
  color: rgba(255,255,255,var(--text-opacity));
  font-size: 0.875rem;
  font-weight: 700;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,-webkit-transform;
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
}

.emotion-0 > div > .acceptButton {
  --bg-opacity: 1;
  background-color: rgba(237,100,166,var(--bg-opacity));
  --text-opacity: 1;
  color: rgba(255,255,255,var(--text-opacity));
  font-weight: 700;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  margin-right: 0.5rem;
  border-style: none;
  cursor: pointer;
}

.emotion-0 > div > .acceptButton:hover {
  --bg-opacity: 1;
  background-color: rgba(184,50,128,var(--bg-opacity));
}

.emotion-0 > div > .declineButton {
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  --text-opacity: 1;
  color: rgba(255,255,255,var(--text-opacity));
  font-weight: 700;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border-style: none;
  cursor: pointer;
}

.emotion-0 > div > .declineButton:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  className="emotion-0 emotion-1"
  style={
    Object {
      "top": "0",
    }
  }
>
  <div
    className=""
    style={Object {}}
  >
    This website uses cookies to enhance the user experience.
  </div>
  <div
    className=""
  >
    <button
      aria-label="Accept cookies"
      className="acceptButton"
      id="rcc-confirm-button"
      onClick={[Function]}
      style={Object {}}
    >
      I understand
    </button>
  </div>
</div>
`;
